e2e/tests/prerelease_checks.robot

```diff
@@ -0,0 +1,118 @@
+*** Settings ***
+Documentation    Pre-release Checks Test Case
+
+Test Tags    pre-release
+
+Library    OperatingSystem
+
+Resource    ../keywords/variables.resource
+Resource    ../keywords/common.resource
+Resource    ../keywords/volume.resource
+Resource    ../keywords/backup.resource
+Resource    ../keywords/engine_image.resource
+Resource    ../keywords/longhorn.resource
+Resource    ../keywords/setting.resource
+
+Test Setup    Set test environment
+Test Teardown    Cleanup test resources
+
+*** Test Cases ***
+Pre-release Checks
+    [Documentation]    Pre-release Checks
+    ...    1. Uninstall existing Longhorn
+    ...    2. Install stable version of Longhorn
+    ...    3. Create volumes and backups
+    ...    4. Upgrade Longhorn
+    ...    5. Upgrade volume engine images
+    ...    6. Trigger replica rebuilding
+    ...    7. Detach/Re-attach volumes
+    ...    8. Restore backups
+    ...    9. Uninstall Longhorn
+    ...    10. Re-install Longhorn back for subsequent tests
+    ${LONGHORN_STABLE_VERSION}=    Get Environment Variable    LONGHORN_STABLE_VERSION    default=''
+    # if Longhorn stable version is provided,
+    # uninstall the existing Longhorn and upgrade Longhorn from the stable version
+    IF    '${LONGHORN_STABLE_VERSION}' != ''
+        Given Set setting deleting-confirmation-flag to true
+        And Uninstall Longhorn
+        And Check Longhorn CRD removed
+
+        And Install Longhorn stable version
+        And Set backupstore
+        And Set up v2 environment
+    END
+
+    And Create volume 0 with    dataEngine=v1
+    And Attach volume 0
+    And Wait for volume 0 healthy
+    And Write data 0 to volume 0
+    And Create backup 0 for volume 0
+
+    # use DATA_ENGINE to control whether to test v2 volumes in the test
+    # because v1.8.x v2 volumes aren't compatible with v1.7.x ones
+    # we have to manually decide whether to include v2 volumes
+    # based on what upgrade path we're testing
+    IF    "${DATA_ENGINE}" == "v2"
+        And Create volume 1 with    dataEngine=v2
+        And Attach volume 1
+        And Wait for volume 1 healthy
+        And Write data 1 to volume 1
+        And Create backup 1 for volume 1
+        And Detach volume 1
+        And Wait for volume 1 detached
+    END
+
+    ${LONGHORN_TRANSIENT_VERSION}=    Get Environment Variable    LONGHORN_TRANSIENT_VERSION    default=''
+    IF    '${LONGHORN_TRANSIENT_VERSION}' != ''
+        When Upgrade Longhorn to transient version
+    END
```

Comment:
If we upgrade from `1.7.x` to `1.8.x` and then to `1.9.x`, after completing this step, will the `volume 0` created in `1.7.x` automatically upgrade volume engine to `1.8.x`?  If not, when we finally upgrade to 1.9.x or master-head, is it likely that the upgrade to v1.9.x will fail?